### PR TITLE
[ENH] Respect SliceEncodingDirection metadata

### LINKED ISSUE #1345: STC ignoring SliceEncodingDirection
Hi.

It seems that stc is not taking SliceEncodingDirection into account. 

I was under the impression that this parameter should alter how SliceTiming is handled.
If my nifti orientation is LPS and the slices where acquired in z direction in descending order (from superior to inferior), SliceEncodingDirection should be 'k-' and SliceTiming something like [0, .05, .1 (...)] correct?
Thanks.

### PR DESCRIPTION
**Pull request by fliem**
## Changes proposed in this pull request
closes #1345 
adds slice encoding direction to stc

depends on [nipype PR#2753](https://github.com/nipy/nipype/pull/2753)

## Documentation that should be reviewed

Not sure...

### CODE REVIEW

**Comment by fliem** [comment]
@effigies thanks for merging the nipype PR. I can't figure out how to get the newest nipype version into fmriprep. Via niworkflows?

**Comment by effigies** [comment]
Once the new nipype is released, we'll incorporate the fix here. We're waiting on one fix in nipype before release (scheduled for Monday anyway).

**Comment by effigies** [comment]
Although, you can pin to the latest nipype commit in requirements.txt, for testing purposes:

```
git+https://github.com/nipy/nipype.git@d83b06d7f8ddb76d4e1033c149eecc966eddb514#egg=nipype
```

**Comment by effigies** [comment]
If you update the nipype dependency to `>=1.1.4`, I think this should be nearly ready.

https://github.com/poldracklab/fmriprep/blob/92786d1456c4098197f6a61e46fde7275f891777/fmriprep/__about__.py#L87-L108

File: fmriprep/workflows/bold/stc.py [hunk 1]
```diff
@@ -71,6 +71,8 @@ def init_bold_stc_wf(metadata, name='bold_stc_wf'):
                     tr='{}s'.format(metadata["RepetitionTime"]),
                     slice_timing=metadata['SliceTiming']),
         name='slice_timing_correction')
+    if 'SliceEncodingDirection' in metadata:
+        slice_timing_correction.inputs.slice_encoding_direction = metadata['SliceEncodingDirection']
```
Comment (effigies) [inline fmriprep/workflows/bold/stc.py]: Maybe we could be as clear (and not hit line length limits) by just adding `slice_encoding_direction=metadata.get('SliceEncodingDirection', 'k')` to the `TShift()` call.

Comment (fliem) [reply fmriprep/workflows/bold/stc.py]: OK. I have tested the previous version and this looks good
* same list order, different direction gives different results (as expected)
* flipped list, flipped direction gives same results (as expected)
* no direction vs "k" gives same results (as expected)

**Comment by effigies** [comment]
Thanks!

**Pull request merged**